Handler/Sites.hs

```diff
@@ -0,0 +1,77 @@
+module Handler.Sites where
+
+import Import
+import Model.Site
+import Helper.Auth
+
+import Control.Monad (void)
+import Yesod.Default.Config (appRoot)
+
+getSitesR :: Handler Html
+getSitesR = do
+    userId <- requireAuthId
+
+    ((result, widget), enctype) <- runFormPost $ siteForm Nothing
+
+    case result of
+        FormSuccess site -> do
+            void $ runDB $ createSite userId site
+            setMessage "Site created!"
+            redirect $ SitesR
+
+        _ -> return ()
+
+    sites <- runDB $ findSites userId
+
+    defaultLayout $ do
+        setTitle "Carnival - Sites"
+        $(widgetFile "sites/index")
+
+postSitesR :: Handler Html
+postSitesR = getSitesR
```

Comment:
Yeah, I know we discussed this yesterday, but after thinking a bit more I think I'll split the functionality. I'm not sure if it's even an idiom anymore, the examples in the Yesod Book no longer do it this way at least.